experimental/dds/tree2/src/test/shared-tree-core/editManager.bench.ts

```diff
@@ -0,0 +1,55 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { benchmark, BenchmarkType } from "@fluid-tools/benchmark";
+import { NoOpChangeRebaser } from "../testChange";
+import {
+	rebaseLocalEditsOverTrunkEdits,
+	rebasePeerEditsOverTrunkEdits,
+} from "./editManagerTestUtils";
+
+describe("EditManager - Bench", () => {
+	interface Scenario {
+		readonly type: BenchmarkType;
+		readonly nbRebased: number;
+		readonly nbTrunk: number;
+	}
+
+	const scenarios: Scenario[] = [
+		{ type: BenchmarkType.Perspective, nbRebased: 1, nbTrunk: 1 },
+		{ type: BenchmarkType.Perspective, nbRebased: 10, nbTrunk: 1 },
+		{ type: BenchmarkType.Perspective, nbRebased: 100, nbTrunk: 1 },
+		{ type: BenchmarkType.Perspective, nbRebased: 1000, nbTrunk: 1 },
+		{ type: BenchmarkType.Perspective, nbRebased: 1, nbTrunk: 10 },
+		{ type: BenchmarkType.Perspective, nbRebased: 1, nbTrunk: 100 },
+		{ type: BenchmarkType.Perspective, nbRebased: 1, nbTrunk: 1000 },
+		{ type: BenchmarkType.Measurement, nbRebased: 100, nbTrunk: 100 },
+	];
+
+	describe("Local commit rebasing", () => {
+		for (const { type, nbRebased, nbTrunk } of scenarios) {
+			benchmark({
+				type,
+				title: `Rebase ${nbRebased} local commits over ${nbTrunk} trunk commits`,
+				benchmarkFn: () => {
+					const rebaser = new NoOpChangeRebaser();
+					rebaseLocalEditsOverTrunkEdits(nbRebased, nbTrunk, rebaser);
```

Comment:
It'd be great if we could profile more precisely by not measuring the creation of the edit manager (i.e. everything in `editManagerFactory`). You might need to use the `benchmarkFnCustom` for that but I think it'd be worth doing.